scripts/migrate_score_override.py

```diff
@@ -16,32 +16,40 @@ def main():
     from openassessment.assessment.models import Assessment, AssessmentPart, StaffWorkflow
     from openassessment.workflow.models import AssessmentWorkflow, AssessmentWorkflowStep
     from student.models import anonymous_id_for_user
-    from submissions.models import Score, ScoreAnnotation, Submission
+    from submissions.models import Score, ScoreSummary, ScoreAnnotation
 
     old_scores = Score.objects.filter(submission__isnull=True, reset=False).order_by('id')
     for score in old_scores:
         try:
             with transaction.atomic():
-                points_override = score.points_earned
-                points_possible = score.points_possible
+                # ScoreSummary is updated on Score saves but for this script we don't want that.
+                # Correct way is to disconnect post_save signal, but since the receiver function
+                # is defined in the class, we can't reference it. Workaround here is to just
+                # prefetch the score summary and resave it to maintain its original field values.
+                score_summary = ScoreSummary.objects.get(student_item=score.student_item)
+
+                # Offset override by 1 second for convenience when sorting db
+                override_date = score.created_at - datetime.timedelta(seconds=1)
+                # Create reset score
+                Score.objects.create(
+                    student_item=score.student_item,
+                    submission=None,
+                    points_earned=0,
+                    points_possible=0,
+                    created_at=override_date,
+                    reset=True,
+                )
 
-                # Update old override with new reset
-                score.points_earned = 0
-                score.points_possible = 0
-                score.reset = True
+                # Update old override with submission from the score preceding it
+                submission = Score.objects.filter(
+                    student_item=score.student_item,
+                    created_at__lt=score.created_at,
+                ).order_by('-created_at')[:1].get().submission
+                score.submission = submission
                 score.save()
 
-                #Grab latest submission with matching student_item
```

Comment:
Was an edge case that the student did a submission after the score override?

Reply:
Yes, basically after the override, the learner state was cleared so the student made another submission after that